### PR TITLE
Dhall.Map: Reflect original key ordering in Ord instance

### PR DESCRIPTION
**Pull request by sjakobi**
This definition is similar to the one in Data.Map:

    instance (Ord k, Ord v) => Ord (Map k v) where
        compare m1 m2 = compare (toAscList m1) (toAscList m2)

### CODE REVIEW

**Comment by sjakobi** [comment]
Ah, maybe I should have expressed myself more clearly. With this change, the original order of the keys gains importance. Previously, this ordering was only considered after comparing the internal `Data.Map`s.

**Comment by EggBaconAndSpam** [comment]
Yes, that makes sense! I got confused because I thought `toList` to be the one from `Data.Map`.

Maybe it would be a good idea to add comments to the typeclass instances clearly describing their behaviour?

File: dhall/src/Dhall/Map.hs [hunk 1]
```diff
@@ -75,8 +75,13 @@ instance (Eq k, Eq v) => Eq (Map k v) where
   Map m1 ks == Map m2 ks' = m1 == m2 && ks == ks'
   {-# INLINABLE (==) #-}
 
+{-|
+>>> fromList [("A",1),("B",2)] < fromList [("B",1),("A",0)]
+True
+-}
 instance (Ord k, Ord v) => Ord (Map k v) where
-  compare (Map mL ksL) (Map mR ksR) = compare mL mR <> compare ksL ksR
+  compare m1 m2 = compare (toList m1) (toList m2)
```
Comment (Gabriel439) [inline dhall/src/Dhall/Map.hs]: I think you could accomplish the same thing more efficiently by a slight modification of the original code:

```suggestion
  compare m1 m2 = compare ksL ksR <> compare mL mR
```

Comment (sjakobi) [reply dhall/src/Dhall/Map.hs]: I don't believe so. Take

```
fromList [('B',1),('A',2)] < fromList [('B',2),('A',1)]
```

That's `True` with my version, but `False` with yours.

Comment (Gabriel439) [reply dhall/src/Dhall/Map.hs]: Ah, you're right.  Never mind, then

**Pull request merged**